### ISSUE TITLE
Baffled by 'open-with' vs '<cr>'

### ISSUE THREAD

**Issue by xxxserxxx** [jarun/nnn#829]
nnn: **v3.5**  
OS: **linux**  

Why do `open-with` and `<enter>` behave differently, and how can I make `<enter>` act like `open-with`+`cli`?

This is something I've fought with before, but it is causing me a lot of trouble with the kakoune integration.  If I `o` a text file and specify the `$EDITOR` command, I get the prompt for CLI/GUI. If I choose **either**, I get the file opened how I want (basically, back in the kakoune instance I left).  However, if I hit `<enter>`, `nnn` opens the file in a new editor instance in a new window.  No combination of environment variables causes `<enter>` to act like `open-with`.

I've tried `!` and `]` shells, and in both the environment looks fine:

```
NNN_PLUG=v:imgviu;t:imgthumb;m:mediainf
NNN_OPTS=cEr
EDITOR=:e
```

That last is something the kakoune nnn plugin sets up, which is *supposed* to send an `open file` command back to the kakoune instance that launched `nnn`.

Note that `NNN_OPENER` is not set in this case.  If I set `NNN_OPENER` also to `:e`, it works as expected.

Is there a page that explains why neither `-c` nor `-E` cause `<enter>` to open files with `$EDITOR`?  What do they do if not that?

*Note*: opened as a bug because this isn't a feature, either.

**Comment by jarun** [comment]
> EDITOR=:e

What is `:e`?

Anyway, to use EDITOR, add the option `-e` and drop `-c`. You should read `nnn -h` carefully.

**Comment by xxxserxxx** [comment]
> What is :e?

In the original report, I mention that:

> That last is something the kakoune nnn plugin sets up, which is supposed to send an open file command back to the kakoune instance that launched nnn.

In other words, it's a command that sends a command back to kakoune through its IPC.  Within the `nnn` instance kakoune launches, I can either `!` or `]` into a shell and type `:e <filename>` and it correctly opens the file in the kakoune instance.  It's part of a kakoune extension called `connect.kak` that simplifies hooking external programs into kakoune, which has a socket-based command-and-control system.  This allows inter-application communications where the window manager or terminal muxer manages windows, rather than the editor (as it is in emacs or neovim).

It's a convenience to avoid hard-coding socket paths into the connectors.

> Anyway, to use EDITOR, add the option -e and drop -c. You should read nnn -h carefully.

Believe me, I did, several times.  The one-sentance description of each parameter is, to me, insufficient to explain the finer details of how these behave. The *[preferably CLI]* wording for `-e` indicates that using the CLI is optional for `nnn`.  Successful communication requires more than just the reader.

That did do the trick, though; I had worked around it by forcing `NNN_OPENER=:e` in the environment, but by switching `NNN_OPTS` to `nEr` I was able to remove `NNN_OPENER` and it's working as I want it to.
